client/packages/system/src/Stock/Components/InventoryAdjustment/InventoryAdjustmentForm.tsx

```diff
@@ -0,0 +1,201 @@
+              />
+            </Box>
+          }
+        />
+      </Box>
+      <Box
+        display="flex"
+        flexDirection="column"
+        gap={1}
+        paddingTop={2}
+        flex={1}
+      >
+        <TextWithLabelRow
+          label={t('label.num-packs')}
+          text={String(stockLine.totalNumberOfPacks)}
+          textProps={{ textAlign: 'end' }}
+          labelProps={{ sx: { textWrap: 'wrap' } }}
+        />
+        <StyledInputRow
+          label={t('label.adjust-by')}
+          Input={
+            <NumericTextInput
+              disabled={!draft.direction}
+              width={160}
+              max={
+                draft.direction === AdjustmentDirectionInput.Reduction
+                  ? stockLine.totalNumberOfPacks
+                  : undefined
+              }
+              value={draft.adjustBy}
+              onChange={adjustBy =>
+                setDraft(state => ({
+                  ...state,
+                  adjustBy: adjustBy ?? 0,
+                  newNumberOfPacks:
+                    state.direction === AdjustmentDirectionInput.Addition
+                      ? stockLine.totalNumberOfPacks + (adjustBy ?? 0)
+                      : stockLine.totalNumberOfPacks - (adjustBy ?? 0),
+                }))
+              }
+            />
+          }
+        />
+        <StyledInputRow
+          label={t('label.new-num-packs')}
+          Input={
+            <NumericTextInput
+              disabled={!draft.direction}
+              width={160}
+              value={draft.newNumberOfPacks}
+              max={
+                draft.direction === AdjustmentDirectionInput.Reduction
+                  ? stockLine.totalNumberOfPacks
+                  : undefined
+              }
+              // // TODO: minimum new # packs when in addition mode
+              // // `min` field doesn't really work... e.g. if current/min is 5,
+              // // user wants to type 20, so starts by typing 2, it will be reset to 5!
+              // // I want a debounced min field or something lol
+              // // maybe error state? (disable ok? or show error message)
+              // // reset to min on lose focus?
+              // min={
+              //   direction === Adjustment.Addition
+              //     ? draft.totalNumberOfPacks
+              //     : undefined
+              // }
+              onChange={newNumPacks =>
+                setDraft(state => ({
+                  ...state,
+                  newNumberOfPacks: newNumPacks ?? 0,
+                  adjustBy: Math.abs(
+                    stockLine.totalNumberOfPacks - (newNumPacks ?? 0)
+                  ),
+                }))
+              }
+            />
+          }
+        />
+        <Box sx={{ display: 'flex', justifyContent: 'end', marginTop: '14px' }}>
+          <DialogButton
+            variant="save"
+            color="primary"
+            disabled={saveDisabled}
+            onClick={save}
+          />
```

Comment:
Thoughts on putting the save button here? Would have preferred to replace the `Ok` modal button with `Save`, but the outer modal component doesn't have access to the tab context so that was looking more complex 👀